svm/src/account_loader.rs

```diff
@@ -235,11 +235,21 @@ impl<'a, CB: TransactionProcessingCallback> AccountLoader<'a, CB> {
     // This is a general purpose function suitable for usage outside initial transaction loading.
     pub(crate) fn load_account(&mut self, account_key: &Pubkey) -> Option<AccountSharedData> {
         match self.do_load(account_key) {
+            // Already seen, from AccountLoader.
+            (Some(account), false) => Some(account),
+            // Exists in accounts-db. Store it in AccountLoader for future loads.
             (Some(account), true) => {
                 self.loaded_accounts.insert(*account_key, account.clone());
                 Some(account)
             }
-            (account, false) => account,
+            // Does not exist in AccountLoader or accounts-db.
+            // Insert a placeholder so we don't need to check again.
+            (None, false) => {
```

Comment:
Subsequent calls to `load_account` for the same non-existent account will always insert a new placeholder. I think the returned `bool` should indicate "whether an accounts-db load was performed" rather than "whether the account came from accounts-db" and then the impossible branch below becomes the insert branch and this `(None, false)` branch just returns `None` as before

Reply:
That means we are missing a test that the same inputs should result in aliasing outputs.